plugins/drill-coverage-plugin/src/adminPartTest/kotlin/CoverageControllerTest.kt

```diff
@@ -0,0 +1,97 @@
+package com.epam.drill.plugins.coverage
+
+import com.epam.drill.plugin.api.end.WsService
+import com.epam.drill.plugin.api.message.DrillMessage
+import kotlinx.coroutines.runBlocking
+import kotlinx.serialization.json.JSON
+import kotlin.test.Test
+import kotlin.test.assertNotNull
+import kotlin.test.assertTrue
+
+class CoverageControllerTest {
+
+    private val ws = WsServiceStub()
+
+    private val coverageController = CoverageController(ws, "test")
+
+    @Test
+    fun `should have empty state after init`() {
+        assertTrue { coverageController.initialClassBytes.isEmpty() }
+        assertTrue { coverageController.javaClasses.isEmpty() }
+        assertTrue { coverageController.prevJavaClasses.isEmpty() }
+    }
+
+    @Test
+    fun `should preserve class data for diff`() {
+        val message = CoverageMessage(CoverageEventType.INIT, "hello")
+
+        coverageController.javaClasses["TestClass"] = JavaClass("TestClass", "TestClass", emptySet())
+
+        runBlocking {
+            coverageController.processData(DrillMessage("", JSON.stringify(CoverageMessage.serializer(), message)))
+        }
+
+        coverageController.apply {
+            assertTrue { initialClassBytes.isEmpty() }
+            assertTrue { javaClasses.isEmpty() }
+            assertTrue { prevJavaClasses.isNotEmpty() }
+        }
+    }
+
+    @Test
+    fun `should add class bytes on receiving a CLASS_BYTES message`() {
+        val clazz = Dummy::class.java
+        val bytes = clazz.readBytes()
+        prepareClasses(clazz)
+
+        coverageController.apply {
+            val path = clazz.path
+            assertNotNull(initialClassBytes[path])
+            assertTrue { initialClassBytes[path]!!.contentEquals(bytes) }
```

Comment:
may be swap this 2 args? "!!" is awful :) bytes.contentEquals(initialClassBytes[path])

Reply:
Agree. The expected value should come first. Fixed it.